apps/cmp.c

```diff
@@ -1311,8 +1312,7 @@ static SSL_CTX *setup_ssl_ctx(OSSL_CMP_CTX *ctx, ENGINE *engine)
     if (opt_tls_trusted != NULL) {
         /* enable and parameterize server hostname/IP address check */
         if (!truststore_set_host_etc(trust_store,
-                                     opt_tls_host != NULL ?
-                                     opt_tls_host : opt_server))
+                                     opt_tls_host != NULL ? opt_tls_host : host))
             /* TODO: is the server host name correct for TLS via proxy? */
```

Comment:
Have you have resolved this TODO comment?  If so, please remove it too.

Reply:
Thanks Pauli for your swift approval!
I also noticed this somewhat related comment. Unfortunately the answer regarding the proxy host name is still open to me.